kristan/src/main/java/com/example/springboot/controller/RestServices.java

```diff
@@ -1,70 +1,139 @@
 package com.example.springboot.controller;
 
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.PatchMapping;
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import com.example.springboot.model.Program;
+import com.example.springboot.model.Student;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.HttpStatus;
+
+import lombok.extern.slf4j.Slf4j;
```

Comment:
Good Day, Hi may I know what this declaration do? it seems essential.

Reply:
this declaration import lombok.extern.slf4j.Slf4j;

Reply:
yes this is for sl4j logging, lombok has several annotation for different logging implementations. Here's the link for more info: https://projectlombok.org/features/log